src/meshes/p4est_mesh.jl

```diff
@@ -497,45 +497,319 @@ function p4est_connectivity_from_hohqmesh_abaqus(meshfile, initial_refinement_le
+    meshfile_p4est_rdy, mesh_polydeg = preprocess_standard_abaqus_for_p4est(meshfile_preproc,
+                                                                            linear_quads,
+                                                                            linear_hexes,
+                                                                            quadratic_quads,
+                                                                            quadratic_hexes,
+                                                                            elements_begin_idx,
+                                                                            sets_begin_idx)
+
     # Create the mesh connectivity using `p4est`
-    connectivity = read_inp_p4est(meshfile, Val(n_dimensions))
+    connectivity = read_inp_p4est(meshfile_p4est_rdy, Val(n_dimensions))
     connectivity_pw = PointerWrapper(connectivity)
 
     # These need to be of the type Int for unsafe_wrap below to work
-    n_trees::Int = connectivity_pw.num_trees[]
+    n_trees::Int = connectivity_pw.num_trees[] # = number elements
     n_vertices::Int = connectivity_pw.num_vertices[]
 
+    # Extract a copy of the element vertices to compute the tree node coordinates
+    # `vertices` store coordinates of all three dimensions (even for the 2D case)
+    # since the Abaqus `.inp` format always stores 3D coordinates.
     vertices = unsafe_wrap(Array, connectivity_pw.vertices, (3, n_vertices))
+
     tree_to_vertex = unsafe_wrap(Array, connectivity_pw.tree_to_vertex,
                                  (2^n_dimensions, n_trees))
 
     basis = LobattoLegendreBasis(RealT, polydeg)
     nodes = basis.nodes
 
+    if mesh_polydeg == 2
+        mesh_nnodes = mesh_polydeg + 1 # = 3
+        # Note: We ASSUME that the additional node between the end-vertices lies 
+        # on the center on that line, such that we can use Chebyshev-Gauss-Lobatto nodes!
+        # For polydeg = 2, we have the 3 nodes [-1, 0, 1] (within the reference element).
+        cheby_nodes, _ = chebyshev_gauss_lobatto_nodes_weights(mesh_nnodes)
+        nodes = SVector{mesh_nnodes}(cheby_nodes)
+    end
+
     tree_node_coordinates = Array{RealT, n_dimensions + 2}(undef, n_dimensions,
                                                            ntuple(_ -> length(nodes),
                                                                   n_dimensions)...,
                                                            n_trees)
-    calc_tree_node_coordinates!(tree_node_coordinates, nodes, mapping, vertices,
-                                tree_to_vertex)
+
+    # No nonlinearity in the mesh. Rely on the mapping function to realize the curvature.                                                           
+    if mesh_polydeg == 1
+        calc_tree_node_coordinates!(tree_node_coordinates, nodes, mapping,
+                                    vertices, tree_to_vertex)
+    else # mesh_polydeg = 2 => Nonlinearity in supplied mesh
```

Comment:
okay, so this logic could readily handle higher order element orders (at least in 2D), right?

Reply:
Yes, so more nodes here should be readily possible. I am not sure, however, if such elements are defined by the Abaqus standard.

Reply:
It is nice that you kept things extensible, but as you found in the ABAQUS docs it might not be possible. I am not sure how extensible the format itself is.